conans/client/rest/file_downloader.py

```diff
@@ -89,21 +99,30 @@ def write_chunks(chunks, path):
         try:
             logger.debug("DOWNLOAD: %s" % url)
             total_length = response.headers.get('content-length') or len(response.content)
-            total_length = int(total_length)
+            total_length = range_start + int(total_length)
             description = "Downloading {}".format(os.path.basename(file_path)) if file_path else None
             progress = progress_bar.Progress(total_length, self._output, description)
 
             chunk_size = 1024 if not file_path else 1024 * 100
             encoding = response.headers.get('content-encoding')
             gzip = (encoding == "gzip")
 
+            # TODO: refactor Progress to allow setting an initial progress
+            progress._processed_size = range_start
+            progress._pb_update(range_start)
+
             written_chunks, total_downloaded_size = write_chunks(
                 progress.update(read_response(chunk_size)),
                 file_path
             )
 
             response.close()
-            if total_downloaded_size != total_length and not gzip:
+            if (
+                total_length > total_downloaded_size > range_start
+                and response.headers.get("accept-ranges") == "bytes"
+            ):
+                written_chunks = self._download_file(url, auth, headers, file_path, try_resume=True)
```

Comment:
This assumes an infinite number of resumed downloads, as long as it keep working, it will keep recursing itself, until it errors or finished, did I understand it correctly?

Reply:
Exactly. Since I wanted to be minimally invasive, I came up with this implementation. For uninterrupted downloads the behavior should not change. Only if the server stops sending data, we have still a chance to make progress with another request as long as we make some progress. Of course that does not cover the case, when a conan command (e.g. install) is invoked a second time.

Reply:
Yes, I am totally fine to not cover the case of consecutive commands, actually when something fails, Conan usually tries to remove stale files, to avoid corrupted things in the cache.